pptxApp.py

```diff
@@ -0,0 +1,158 @@
+    """
+    this method receives a path for a pptx presentation, checks if the path is found in the operating system, if so,
+    parses the data to slides. the method, returns a list of explanations.
+    :param: presentation_path: path of a power-point presentation (String)
+    :return: list of explanations (List of strings)
+    """
+    if not os.path.isfile(presentation_path):  # check if path is available
+        print(f"{ERROR_MESSAGE} the path you provided does not exist.")
+        return []
+
+    prs = Presentation(presentation_path)
+    explanations = []
+    for slide in prs.slides:
+        explanation = await parse_slide_of_pptx(slide)  # call method to parse data in each slide
+        explanations.append(explanation)
+    return explanations
+
+
+# --------------------------------------------------------------------------------------------------------------------
+async def parse_slide_of_pptx(slide):
+    """
+    this method receives a single slide, the method parses the data on that slide, it calls another method to get the
+    response and return it to the parse_presentation method. it throws an error if an exception occurred.
+    :param: slide: a single slide from the power-point (slide object)
+    :return: the explanation if the processing went well, an error, otherwise. (List of Strings)
+    """
+    try:
+        response = await request_completion(slide)  # call a function to get explanations on each slide.
+        return response
+    except Exception as error:  # if an error occurred, display a suitable message to user.
+        error_message = f"{ERROR_MESSAGE} Error processing slide: {str(error)}"
+        return error_message
+
+
+# --------------------------------------------------------------------------------------------------------------------
+def parse_text_of_slide(slide):
+    """
+    this method receives a slide, and extracts all the extractable text found in that slide. In addition, it cleans
+    the data using the strip function.
+    :param: slide: a single slide from the power-point (Slide Object)
+    :return: explanation response (String)
+    """
+    slide_text = []
+    for shape in slide.shapes:
+        if not shape.has_text_frame:
+            continue
+        for paragraph in shape.text_frame.paragraphs:
+            for run in paragraph.runs:
+                slide_text.append(run.text.strip())
+    return " ".join(slide_text)
+
+
+# --------------------------------------------------------------------------------------------------------------------
+async def request_completion(slide):
+    """
+    this method receives a slide object, sends a request to the openai API asking the server to explain the content
+    of that slide, the method returns the response from the API
+    :param: slide: slide: a single slide from the power-point (Slide Object)
+    :return: response of the API
+    """
+    CONTENT.append({"role": "user", "content": parse_text_of_slide(slide)})
```

Comment:
That's bad practice using a constant as a global variable.
First, that means it's not a constant, second, it can be implemented in much better ways - class and so on.